### DISCUSSION TITLE
Problem in writeBinary() function

### DISCUSSION THREAD

**Discussion by HIMANSHU6001** [donmccurdy/glTF-Transform#1486] [Q&A]
Hello, I have been struggling with this issue since long ago, I am trying to draco compress the meshes in the scene and convert them to a Unit8Array which i will be storing later in database the code is like this :- 
```javascript
const compressAndExport = async (scene) => {
    setShowSaveModal(true);
    exporter.parse(
      scene,
      async function (glb) {
        const unit8array = new Uint8Array(glb);
        console.log("Before compression = ", unit8array);
        const glbfile = await io.readBinary(unit8array);
        glbfile
          .createExtension(KHRDracoMeshCompression)
          .setRequired(true)
          .setEncoderOptions({
            method: KHRDracoMeshCompression.EncoderMethod.SEQUENTIAL,
            encodeSpeed: 5,
            decodeSpeed: 5,
          });
        await glbfile.transform(weld(), quantize(), dedup());
        console.log("After compression = ", glbfile);

        const BinaryData = await io.writeBinary(glbfile);
        console.log("BinaryData = ", BinaryData);
        // setModelData(BinaryData);
      },
      function (error) {
        console.log("An error happened", error);
      },
      options
    );
  };
```

These are the last Few logs of the console :- 
  

Before compression =  Uint8Array(44248) [103, 108, 84, 70, 2, 0, 0, 0, 216, 172, 0, 0, 32, 54, 0, 0, 74, 83, 79, 78, 123, 34, 97, 115, 115, 101, 116, 34, 58, 123, 34, 118, 101, 114, 115, 105, 111, 110, 34, 58, 34, 50, 46, 48, 34, 44, 34, 103, 101, 110, 101, 114, 97, 116, 111, 114, 34, 58, 34, 84, 72, 82, 69, 69, 46, 71, 76, 84, 70, 69, 120, 112, 111, 114, 116, 101, 114, 32, 114, 49, 54, 55, 34, 125, 44, 34, 115, 99, 101, 110, 101, 115, 34, 58, 91, 123, 34, 110, 97, 109, …]
EditorPage.jsx?t=1724499238005:126 After compression =  _Document {_graph: Graph, _root: Root, _logger: _Logger}
chunk-HZOFJH27.js?v=409da319:1171 Uncaught (in promise) TypeError: encoderModule.MeshBuilder is not a constructor
    at encodeGeometry (chunk-HZOFJH27.js?v=409da319:1171:19)
    at KHRDracoMeshCompression.prewrite (chunk-HZOFJH27.js?v=409da319:1381:23)
    at chunk-SL57VI3R.js?v=409da319:4783:132
    at Array.forEach (<anonymous>)
    at GLTFWriter.write (chunk-SL57VI3R.js?v=409da319:4783:99)
    at WebIO.writeJSON (chunk-SL57VI3R.js?v=409da319:5223:23)
    at WebIO.writeBinary (chunk-SL57VI3R.js?v=409da319:5237:20)
    at EditorPage.jsx?t=1724499238005:127:37

**Comment by donmccurdy** [comment]
Hi @HIMANSHU6001 – it sounds like Draco is not properly installed in your application. Is this code running in node.js, or on the web? If the web, installing the Draco encoder's WASM dependencies is a bit tricky, see https://stackoverflow.com/questions/66975407/convert-glb-to-draco-glb-front-end/66978236#66978236.
